tests/stress/deployments.py

```diff
@@ -0,0 +1,761 @@
+        )
+        memgraph.execute(
+            'REGISTER INSTANCE instance_2 WITH CONFIG {"bolt_server": "localhost:7688", "management_server": "localhost:13012", "replication_server": "localhost:10002"};'
+        )
+        memgraph.execute(
+            'REGISTER INSTANCE instance_3 WITH CONFIG {"bolt_server": "localhost:7689", "management_server": "localhost:13013", "replication_server": "localhost:10003"};'
+        )
+        memgraph.execute("SET INSTANCE instance_1 TO MAIN;")
+
+    def stop_memgraph(self) -> None:
+        for proc in self._data_proc + self._coord_proc:
+            proc.terminate()
+            ret_mg = proc.wait()
+            if ret_mg != 0:
+                raise Exception("Memgraph binary returned non-zero ({})!".format(ret_mg))
+
+    def cleanup(self):
+        """Terminate all processes and clean up data directories."""
+        # Stop running processes
+        for proc in self._data_proc + self._coord_proc:
+            if proc.poll() is None:
+                proc.terminate()
+                proc.wait()
+
+        # Remove data directories and log files from the build directory
+        for data_config in self._data_config:
+            data_dir = os.path.join(BUILD_DIR, f"mg_data_{data_config['id']}")
+            log_file = os.path.join(BUILD_DIR, f"mg_data_{data_config['id']}.log")
+            if os.path.exists(data_dir):
+                shutil.rmtree(data_dir)
+                print(f"Removed data directory: {data_dir}")
+            if os.path.exists(log_file):
+                os.remove(log_file)
+                print(f"Removed log file: {log_file}")
+
+        for coord_config in self._coord_config:
+            coord_dir = os.path.join(BUILD_DIR, f"mg_coord_{coord_config['id']}")
+            log_file = os.path.join(BUILD_DIR, f"mg_coord_{coord_config['id']}.log")
+            if os.path.exists(coord_dir):
+                shutil.rmtree(coord_dir)
+                print(f"Removed coordinator directory: {coord_dir}")
+            if os.path.exists(log_file):
+                os.remove(log_file)
+                print(f"Removed coordinator log file: {log_file}")
+
+    def wait_for_server(self, port) -> None:
+        cmd = ["nc", "-z", "-w", "1", "127.0.0.1", str(port)]
+        while subprocess.call(cmd) != 0:
+            time.sleep(0.5)
+        time.sleep(2)
+
+    def _get_data_instance_ha_flags(self, data_config):
+        return [
+            f"--management_port={data_config['management-port']}",
+            f"--bolt-port={data_config['bolt-port']}",
+            f"--monitoring-port={data_config['monitoring-port']}",
+            f"--metrics-port={data_config['metrics-port']}",
+            f"--data-directory=mg_data_{data_config['id']}",
+            f"--log-file=mg_data_{data_config['id']}.log",
+            f"--replication-restore-state-on-startup=false",
+            f"--data-recovery-on-startup=false",
```

Comment:
I think the very important point in HA is restarting instances and managing that. Maybe you don't need this now but I think part of stress testing HA must be restarting instances.